src/Logger.js

```diff
@@ -0,0 +1,105 @@
+const pino = require("pino");
+
+/**
+ * Wrapper class around pino package to add some extra functionality.
+ */
+module.exports = class Logger {
+    /**
+     * @param {string} loglevel  Accepted values: fatal | error | warn | info | debug | trace
+     * @param {object} context   Common props to add to all logged messages.
+     */
+    constructor(loglevel = "info", context = {}) {
+        this._logger = pino({
+            messageKey: "message",
+            base:       null,
+            timestamp:  true,
+            level:      loglevel
+        });
+        this.setContext(context);
+    }
+
+    /**
+     * Sets the pino-compatible log level to be used from this point forward.
+     *
+     * @param {string} loglevel  Accepted values: fatal | error | warn | info | debug | trace
+     */
+    setLogLevel(loglevel) {
+        this._logger.level = loglevel;
+    }
+
+    /**
+     * Sets common props that will be added to all logs from this point forward.
+     * Useful for things like request_id, type/namespace/service, etc.
+     *
+     * @param {object} context   Common props to add to all logged messages.
+     */
+    setContext(context) {
+        this._context = context;
+    }
+
+    /**
+     * Adds/overwrites common props that will be added to all logs from this point forward.
+     * Useful for things like request_id, type/namespace/service, etc.
+     *
+     * @param {object} context   Common props to add to all logged messages.
+     */
+    addContext(context) {
+        this._context = {...this._context, ...context};
+    }
+
+    /**
+     * Creates a child logger from the current one, inherting all of parent logger context.
+     *
+     * @param {object} context   Common props to add to current context.
+     */
+    child(context) {
+        return new Logger(this._logger.level, {...this._context, ...context});
+    }
+
+    info(message, props = {}) {
+        this._log("info", message, props);
+    }
+
+    warn(message, props = {}) {
+        this._log("warn", message, props);
+    }
+
+    error(message, props = {}) {
```

Comment:
One last comment that just come up to my mind. Don't we have places in which we also want to pass props AND an instance of the error? If yes, that won't work with this approach and you need to add a specific `error` param in addition to `props`

Reply:
So far I haven't detected any case in adserver or adstitcher. Do you think we need to add support for this? In that case, though, the first parameter should always be an `Error` or `null/undefined` and then the second, optional props. What do you think?

Reply:
> In that case, though, the first parameter should always be an Error or null/undefined and then the second, optional props. What do you think?

Yeah, I agree, it makes more sense for the error to be first. In this case, up to you about doing it now or later when there'll be the need

Reply:
I think it's better for now to keep it this way and if needed change it later